### PR TITLE
[windows] warmup Azure CLI

### PR DESCRIPTION
**Pull request by ilia-shipitsin**
# Description

warmup Azure CLI during image generation to improve first run timing

#### Related issue:

https://github.com/actions/runner-images/issues/8296

## Check list
- [x] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

**Review by jessehouwing — APPROVED** [review]

**Comment by jessehouwing** [comment]
Replaces the part of this pull request that performs the redirection of the `.azure` folder and the warmup of `az`:

 * https://github.com/actions/runner-images/pull/8294

File: images/win/scripts/Installers/Install-AzureCli.ps1 [hunk 1]
```diff
@@ -3,13 +3,32 @@
...
+# Warm-up Azure CLI
+
+Write-Host "Warmup 'az'"
 
-[Environment]::SetEnvironmentVariable("AZURE_EXTENSION_DIR", $azureCliExtensionPath, [System.EnvironmentVariableTarget]::Machine)
+az --help
+if ($LASTEXITCODE -ne 0)
+{
+   throw "Command 'az --help' failed"
+}
```
Comment (vpolikarpov-akvelon) [inline images/win/scripts/Installers/Install-AzureCli.ps1:32]: Don't we have tests that do the same?

Comment (ilia-shipitsin) [reply images/win/scripts/Installers/Install-AzureCli.ps1:32]: even if we do, it would be a leaked abstraction.
tests do testing, warmup does warmup

Comment (jessehouwing) [reply images/win/scripts/Installers/Install-AzureCli.ps1:32]: Current test does `az --version`. Which is nice too, as it causes the caching of the version information on disk (as `az --version` does an update check and caches the result).

Comment (ilia-shipitsin) [reply images/win/scripts/Installers/Install-AzureCli.ps1:32]: someone once may want to change tests behavior. I would not like to depend on those changes, thus dedicated warm up makes sense (even if it's identical)

Comment (vpolikarpov-akvelon) [reply images/win/scripts/Installers/Install-AzureCli.ps1:32]: Ok. I would only hide az's output and put braces according to1TBS rule.

Comment (ilia-shipitsin) [reply images/win/scripts/Installers/Install-AzureCli.ps1:32]: I used VS Code with whatever we have in .vscode, I wonder why do you think code is not compliant

**Review by mikhailkoliada — APPROVED** [review]

**Pull request merged**